default.nix

```diff
@@ -0,0 +1,31 @@
+{ mkDerivation, aeson, ansi-terminal, async, base, directory
+, filepath, formatting, gitrev, hspec, hspec-discover, mtl
+, optparse-applicative, path, path-io, process, QuickCheck
+, silently, stdenv, stm, temporary, time, transformers
+, typed-process, yaml, git, zlib, zsh
+}:
+mkDerivation {
+  pname = "hapistrano";
+  version = "0.3.10.0";
+  src = ./.;
+  isLibrary = true;
+  isExecutable = true;
+  enableSeparateDataOutput = true;
+  libraryHaskellDepends = [
+    aeson ansi-terminal base filepath formatting gitrev mtl path
+    process stm time transformers typed-process
+  ];
+  executableHaskellDepends = [
+    aeson async base formatting gitrev optparse-applicative path
+    path-io stm yaml
+  ];
+  testHaskellDepends = [
+    base directory filepath hspec mtl path path-io process QuickCheck
+    silently temporary
+  ];
+  testSystemDepends = [ git zlib zsh ];
```

Comment:
I have a question. Since many of the commands are actually a wrapper for `ssh` and `scp`, do we need to add them as dependencies for the package?

Reply:
Those dependencies are for testing, as when nix is building the package it creates a nix-shell (environment) with those packages available. But, when you execute the program with "hap" after installation the binaries available will be the ones on your PATH.

Reply:
So I don't think adding ssh and scp is necessary, unless a test uses those commands.